From fe0d49ff6277201dd549a236b67beed9f6831dd6 Mon Sep 17 00:00:00 2001
From: Hanlu Li <heuleehanlu@gmail.com>
Date: Thu, 3 Apr 2025 15:02:26 +0800
Subject: [PATCH 1030/1058] LATX, fix: Fix an build error

This patch to fix failure with build64-dbg.sh.
Build now works.

Signed-off-by: Hanlu Li <heuleehanlu@gmail.com>
---
 linux-user/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/linux-user/main.c b/linux-user/main.c
index bf19b0d71e..4c0fc1e02f 100644
--- a/linux-user/main.c
+++ b/linux-user/main.c
@@ -1288,8 +1288,8 @@ int main(int argc, char **argv, char **envp)
                     PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
                 if (addr != MAP_FAILED) {
                     munmap(addr, qemu_host_page_size);
-                    assert(!(addr > KERNEL_CONFIG_LSM_MMAP_MIN_ADDR &&
-                            addr > mmap_min_addr));
+                    assert(!((unsigned long)addr > KERNEL_CONFIG_LSM_MMAP_MIN_ADDR &&
+                            (unsigned long)addr > mmap_min_addr));
                     mmap_min_addr = (unsigned long)addr;
                     qemu_log_mask(CPU_LOG_PAGE, "host mmap_min_addr=0x%lx\n",
                                   mmap_min_addr);
-- 
2.49.0

